File: dart/bin/smoke.dart
import 'dart:io';

import 'package:apialerts/apialerts.dart';

Future<void> main() async {
  final apiKey = Platform.environment['APIALERTS_API_KEY'];
  if (apiKey == null || apiKey.isEmpty) {
    stderr.writeln('Error: APIALERTS_API_KEY is not set');
    exit(1);
  }

  ApiAlerts.configure(apiKey);

  final result = await ApiAlerts.sendAsync(const Event(
    message: 'Published smoke - Dart',
    channel: 'testing',
    event: 'ci.sdk.smoke.dart',
    title: 'Published Package OK',
    tags: ['smoke', 'dart', 'published'],
    link: 'https://github.com/apialerts/apialerts-dart',
    data: {'language': 'dart'},
  ));

  if (!result.success) {
    stderr.writeln('FAIL: ${result.error}');
    exit(1);
  }
  stdout.writeln('OK: dart sent to ${result.workspace} (${result.channel})');
}

File: ruby/smoke.rb
require 'apialerts'

api_key = ENV['APIALERTS_API_KEY']
if api_key.nil? || api_key.empty?
  warn 'Error: APIALERTS_API_KEY is not set'
  exit 1
end

ApiAlerts.configure(api_key)

result = ApiAlerts.send_async(ApiAlerts::Event.new(
  message: 'Published smoke - Ruby',
  channel: 'testing',
  event:   'ci.sdk.smoke.ruby',
  title:   'Published Package OK',
  tags:    %w[smoke ruby published],
  link:    'https://github.com/apialerts/apialerts-ruby',
  data:    { language: 'ruby' }
))

unless result.success?
  warn "FAIL: #{result.error}"
  exit 1
end

puts "OK: ruby sent to #{result.workspace} (#{result.channel})"
